apps/settings/components/coupon_code/view.coffee

```diff
@@ -0,0 +1,46 @@
+Backbone = require 'backbone'
+Promise = require 'bluebird-q'
+{ debounce } = require 'underscore'
+template = -> require('./index.jade') arguments...
+
+module.exports = class CouponCodeView extends Backbone.View
+  els: {}
+
+  events:
+    'input input': 'onInput'
+
+  initialize: ->
+    @listenTo @model, 'change:description', @render
+    @listenTo @model, 'change:code', @render
+    @listenTo @model, 'change:code', debounce(@checkCode, 500)
+
+  onInput: ->
+    code = @els.input.val()
```

Comment:
Look above. Just an object. Just my own stylistic quirks. I prefer this over `@$input`.